flow360/component/flow360_params/unit_system.py

```diff
@@ -0,0 +1,589 @@
+    """:class: Flow360LengthUnit"""
+
+    dimension_type = LengthType
+    unit_name = "flow360_length_unit"
+
+
+class Flow360MassUnit(_Flow360BaseUnit):
+    """:class: Flow360MassUnit"""
+
+    dimension_type = MassType
+    unit_name = "flow360_mass_unit"
+
+
+class Flow360TimeUnit(_Flow360BaseUnit):
+    """:class: Flow360TimeUnit"""
+
+    dimension_type = TimeType
+    unit_name = "flow360_time_unit"
+
+
+class Flow360TemperatureUnit(_Flow360BaseUnit):
+    """:class: Flow360TemperatureUnit"""
+
+    dimension_type = TemperatureType
+    unit_name = "flow360_temperature_unit"
+
+
+class Flow360VelocityUnit(_Flow360BaseUnit):
+    """:class: Flow360VelocityUnit"""
+
+    dimension_type = VelocityType
+    unit_name = "flow360_velocity_unit"
+
+
+class Flow360AreaUnit(_Flow360BaseUnit):
+    """:class: Flow360AreaUnit"""
+
+    dimension_type = AreaType
+    unit_name = "flow360_area_unit"
+
+
+class Flow360ForceUnit(_Flow360BaseUnit):
+    """:class: Flow360ForceUnit"""
+
+    dimension_type = ForceType
+    unit_name = "flow360_force_unit"
+
+
+class Flow360PressureUnit(_Flow360BaseUnit):
+    """:class: Flow360PressureUnit"""
+
+    dimension_type = PressureType
+    unit_name = "flow360_pressure_unit"
+
+
+class Flow360DensityUnit(_Flow360BaseUnit):
+    """:class: Flow360DensityUnit"""
+
+    dimension_type = DensityType
+    unit_name = "flow360_density_unit"
+
+
+class Flow360ViscosityUnit(_Flow360BaseUnit):
+    """:class: Flow360ViscosityUnit"""
+
+    dimension_type = ViscosityType
+    unit_name = "flow360_viscosity_unit"
+
+
+class Flow360AngularVelocityUnit(_Flow360BaseUnit):
+    """:class: Flow360AngularVelocityUnit"""
+
+    dimension_type = AngularVelocityType
+    unit_name = "flow360_angular_velocity_unit"
+
+
+_lock = Lock()
+
+
+class UnitSystem(pd.BaseModel):
+    """
+    :class: Customizable unit system containing definitions for most atomic and complex dimensions.
+    """
+
+    mass: MassType = pd.Field()
+    length: LengthType = pd.Field()
+    time: TimeType = pd.Field()
+    temperature: TemperatureType = pd.Field()
+    velocity: VelocityType = pd.Field()
+    area: AreaType = pd.Field()
+    force: ForceType = pd.Field()
+    pressure: PressureType = pd.Field()
+    density: DensityType = pd.Field()
+    viscosity: ViscosityType = pd.Field()
+    angular_velocity: AngularVelocityType = pd.Field()
+
+    @staticmethod
+    def __get_unit(system, dim_name, unit):
+        if unit is not None:
+            return unit
+        if system is not None:
+            if system == "SI":
+                return _SI_system[dim_name]
+            if system == "CGS":
+                return _CGS_system[dim_name]
+            if system == "Imperial":
+                return _imperial_system[dim_name]
+            if system == "Flow360":
+                return _flow360_system[dim_name]
+        return None
+
+    def __init__(self, **kwargs):
+        base_system = kwargs.get("base_system")
+
+        dim_names = [
```

Comment:
What is the purpose of strings for dimension names here?

It is preferable to use enums or something similar to define properties

Python has reflection so you can convert enums to strings easily.

Reply:
This is true, changed the argument type to string in 48ebb7c